File: WinterCodeRun/2. Калибровка магического альтиметра/solution.py
import sys


def main():
    nums = [int(sys.stdin.readline().rstrip()) for _ in range(10)]
    sums = {0}
    for num in nums:
        curr = set()
        for sum in sums:
            curr.add(sum + num)
        sums.update(curr)
    diff = 101
    res = -1
    for sum in sums:
        curr_diff = abs(100 - sum)
        if curr_diff == 0:
            diff = curr_diff
            res = sum
        elif curr_diff < diff:
            diff = curr_diff
            res = sum
        elif curr_diff == diff and sum > res:
            res = sum
            diff = curr_diff
    sys.stdout.write(str(res))


if __name__ == '__main__':
    main()

File: WinterCodeRun/2. Калибровка магического альтиметра/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReaderSize(os.Stdin, 1<<20)

	nums := make([]int, 0, 10)
	for i := 0; i < 10; i++ {
		line, _ := reader.ReadString('\n')
		num, _ := strconv.Atoi(strings.Fields(line)[0])
		nums = append(nums, num)
	}

	sums := make([]int, 1, 1024)
	for _, num := range nums {
		currSum := make([]int, 100)
		for _, sum := range sums {
			currSum = append(currSum, sum+num)
		}
		sums = append(sums, currSum...)
	}
	diff := 101
	res := -1
	for _, sum := range sums {
		currDiff := 0
		if 100-sum > 0 {
			currDiff = 100 - sum
		} else {
			currDiff = (100 - sum) * -1
		}
		if currDiff == 0 {
			diff = currDiff
			res = sum
		} else if currDiff < diff {
			diff = currDiff
			res = sum
		} else if currDiff == diff && sum > res {
			diff = currDiff
			res = sum
		}
	}
	fmt.Printf("%d\n", res)
}
